From f2b3e2faa57b44952ec7f368c4a1eb0f04f35455 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Thu, 26 Dec 2024 16:30:43 +0800
Subject: [PATCH 049/119] mtk: mt76: mt7996: use GFP_DMA32 for page_pool

Use page_pool_alloc_frag to send GFP_DMA32 for page pool. Without this patch,
it cannot support 64-bit environment.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt76.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mt76.h b/mt76.h
index 9dd40c28..f2b6cd6d 100644
--- a/mt76.h
+++ b/mt76.h
@@ -2084,9 +2084,10 @@ static inline void mt76_put_page_pool_buf(void *buf, bool allow_direct)
 static inline void *
 mt76_get_page_pool_buf(struct mt76_queue *q, u32 *offset, u32 size)
 {
+	gfp_t gfp = (GFP_ATOMIC | __GFP_NOWARN | GFP_DMA32);
 	struct page *page;
 
-	page = page_pool_dev_alloc_frag(q->page_pool, offset, size);
+	page = page_pool_alloc_frag(q->page_pool, offset, size, gfp);
 	if (!page)
 		return NULL;
 
-- 
2.45.2

